dev-tools/releng/src/main.rs

```diff
@@ -0,0 +1,669 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod cmd;
+mod hubris;
+mod job;
+mod tuf;
+
+use std::sync::Arc;
+use std::time::Duration;
+use std::time::Instant;
+
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use chrono::Utc;
+use clap::Parser;
+use fs_err::tokio as fs;
+use omicron_zone_package::config::Config;
+use once_cell::sync::Lazy;
+use semver::Version;
+use slog::debug;
+use slog::error;
+use slog::info;
+use slog::Drain;
+use slog::Logger;
+use slog_term::FullFormat;
+use slog_term::TermDecorator;
+use tokio::process::Command;
+use tokio::sync::Semaphore;
+
+use crate::cmd::CommandExt;
+use crate::job::Jobs;
+
+/// The base version we're currently building. Build information is appended to
+/// this later on.
+///
+/// Under current policy, each new release is a major version bump, and
+/// generally referred to only by the major version (e.g. 8.0.0 is referred
+/// to as "v8", "version 8", or "release 8" to customers). The use of semantic
+/// versioning is mostly to hedge for perhaps wanting something more granular in
+/// the future.
+const BASE_VERSION: Version = Version::new(8, 0, 0);
```

Comment:
Is this going to become the sole source of truth for the system version?

Reply:
That's the plan, until we eventually find a better one. (The former source of truth was `.github/buildomat/jobs/package.sh`, which generated a `version.txt` file.)

Reply:
I guess the main alternative is that this lives somewhere in a toml file? I think it's fine as is though.